channeldb/invoices.go

```diff
@@ -152,6 +162,13 @@ type InvoiceRef struct {
 	// currently indexed by payment hash. This value will be used as a
 	// fallback when no payment address is known.
 	payHash lntypes.Hash
+
+	// payAddr is the payment addr of the target invoice. Newer invoices
+	// (0.11 and up) are indexed by payment address in addition to payment
+	// hash, but pre 0.8 invoices do not have one at all. When this value is
+	// known it will be used as the primary identifier, falling back to
+	// payHash if no value is known.
+	payAddr *[32]byte
```

Comment:
Not necessarily for this PR, but I can see an advantage of making an `lntypes.Hash` style datatype for payment address, especially if we are going to use it a lot

Reply:
yes i've been thinking maybe we should, but held off because we don't use it for anything other than comparing equality. definitely something we can look into tho, it would be nice if they were unified